### PR TITLE
Updating WYS (wys_api.py) doc

### PR DESCRIPTION
**Pull request by tahaislam**
## What this pull request accomplishes:

- Update the documentation of the `wys_api.py` script to match the updated code

## Issue(s) this solves:

- #400

### CODE REVIEW

File: wys/api/README.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Watch Your Speed Sign API
 
+## Overview
+
+!['A sign mounted on a pole with the words "Your Speed" and underneath a digital sign displaying "31"'](https://www.toronto.ca/wp-content/uploads/2018/09/9878-landscape-mwysp2-e1538064432616-1024x338.jpg)
+
+The city has installed [Watch Your Speed Signs](https://www.toronto.ca/services-payments/streets-parking-transportation/road-safety/vision-zero/safety-initiatives/initiatives/watch-your-speed-program/) that display the speed a vehicle is travelling at and flashes if the vehicle is travelling over the speed limit. Installation of the sign was done as part of 3 programs: the normal watch your speed sign program, mobile watch your speed which has signs mounted on trailers that move to a different location every few weeks, and school watch your speed which has signs installed at high priority schools. As part of the [Vision Zero Road Safety Plan](https://www.toronto.ca/services-payments/streets-parking-transportation/road-safety/vision-zero/), these signs aim to reduce speeding.
+
```
Comment (ShahrzadBorjian2) [inline wys/api/README.md:8]: Can you also add a link to the general WYS readme?

File: wys/api/README.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
 This API script can grab data from each watch your speed sign the city has. It can perform some functionality that the [streetsoncloud portal](www.streetsoncloud.com) has, mainly pulling speed and volume data. The API supports more calls including setting/getting the schedule, setting/getting the messages each sign displays and other calls.
```
Comment (ShahrzadBorjian2) [inline wys/api/README.md:9]: Can you provide more info on streetsoncloud?

File: wys/api/README.md [hunk 3]
```diff
@@ -10,90 +16,85 @@ The script can request data on any day specified by the API call.
...
+   3. Aggregate speed counts by calling the `plpgsql` function `wys.aggregate_speed_counts_one_hour_5kph()`
+4. Update the `wys.locations` table with the changes in the direction and/or location (moved for more than 100 meters) of any existing sign and insert the locations of the new signs as well.
+5. Update the `wys.sign_schedules_list` table, if needed
 
 ## PostgreSQL Processing
 
 ### Data Tables
 
-The data is inserted into `wys.raw_data`. Data from the API is already pre-aggregated into roughly 5 minute bins.
+The data in `wys.raw_data` table is pre-aggregated into roughly 5 minute bins, and the table has the following columns:
```
Comment (ShahrzadBorjian2) [inline wys/api/README.md:57]: Why roughly?

File: wys/api/README.md [hunk 4]
```diff
@@ -10,90 +16,85 @@ The script can request data on any day specified by the API call.
...
 
 ### Inconsistent time bins
 
-The API only gives data on non-regular (but uniform) 5 minute bins, meaning data is given at 3:44, 3:49 etc. This change has been accomodated in the tables, and for the 15 minute aggregation tables.
+The API provides data on non-regular (but uniform) 5 minute bins, i.e., at 3:44, 3:49, etc. This change has been accomodated in the tables, and for the 15 minute aggregation tables.
 
 ## Calls and Input Parameters
 
-The script uses the `get statistics` call to get the volume and speed data. It will return the count of vehicles for each speed recorded by the sign in roughly 5 minute aggregate bins.
+The WYS data puller script is called by an Airflow DAG that runs daily. It collects the count of vehicles for each speed recorded by the sign in roughly 5 minute aggregate bins.
```
Comment (ShahrzadBorjian2) [inline wys/api/README.md:29]: Same question, why roughly 5 min?

File: wys/api/README.md (hunk 2 above)
Comment (ShahrzadBorjian2) [inline wys/api/README.md:9]: Can you provide more info on streetsoncloud?

File: wys/api/README.md (hunk 1 above)
Comment (ShahrzadBorjian2) [inline wys/api/README.md:8]: Can you also add a link to the general WYS readme?

File: wys/api/README.md [hunk 5]
```diff
@@ -10,90 +16,85 @@ The script can request data on any day specified by the API call.
...
+|`api_id`|integer|ID used for the API, and unique for each sign (most mobile signs change locations every three weeks)|1967|
+|`address`|text|Address of the sign|1577 Bloor Street West|
+|`sign_name`|text|Name of the sign. May include address + serial number, the ward name for the Mobile WYSP, or school name for Schools WYSP|Dundas St W SB 16101191|
+|`dir`|text|Direction of the flow of traffic|NB|
+|`start_date`|date|First date of valid data|2018-11-28|
+|`loc`|text|The coordinates of the sign|(43.666115,-79.370164)
+|`id`|integer|Unique ID and primary key of the table|1|
+|`geom`|geometry|The location of the sign calculated from `loc`||
 
 ## Quality Checks
```
Comment (ShahrzadBorjian2) [inline wys/api/README.md:99]: Can you add a link to the notebook?

File: wys/api/README.md [hunk 6]
```diff
@@ -10,90 +16,85 @@ The script can request data on any day specified by the API call.
...
-|------|------|-------|------|
-`api_id`|integer|ID used for the API, and unique identifier for the `locations` table|1967
-`address`|text|Address of the sign|1577 Bloor Street West
-`sign_name`|text|Name of the sign. May include address + serial number, the ward name for the Mobile WYSP, or school name for Schools WYSP|Dundas St W SB 16101191
-`dir`|text|Direction of the flow of traffic|NB
-
-### Views
-
-`report_dates` contains the list of days, for each sign, where there is more than 10 hours of data between 6:00 and 22:00.
+`locations` contains each ID used for the API, direction of traffic along which the sign is collecting data, and address and name of the sign. Some signs may have information missing from any one of the fields.
```
Comment (ShahrzadBorjian2) [inline wys/api/README.md:86]: How about locations of  mobile signs? Are they pulled from an excel file?

File: wys/api/README.md [hunk 7]
```diff
@@ -10,90 +16,85 @@ The script can request data on any day specified by the API call.
...
+|`speed_counts_agg_5kph_id`|bigint|A unique identifier for the `speed_counts_agg_5kph` table|2655075|
+|`api_id`|integer|ID used for the API, and unique identifier for the `locations` table|1967|
+|`datetime_bin`|timestamp|Start time of the 1 houraggregated bin|2018-10-29 10:00:00|
+|`speed_id`|integer|A unique identifier for the 5 kph speed bin in the `speed_bins` table|5|
+|`volume`|integer|Number of vehicles in datetime_bin/api_id/speed bin combination|7|
 
 ### Lookup Tables
 
-`speed_bins` is a lookup table containing all the 5km/h speed bin. Bin number 25 contains any speed over 120km/h.
+`speed_bins_old` is a lookup table containing all the 5km/h speed bin. Bin number 21 contains any speed over 100km/h.
```
Comment (ShahrzadBorjian2) [inline wys/api/README.md:79]: What is the difference between speed_bins and speed_bin_old?  Only # of bins?
Also I cannot find speed_bin in postgres.

File: wys/api/README.md [hunk 8]
```diff
@@ -10,90 +16,85 @@ The script can request data on any day specified by the API call.
...
-|Field name|Data type|Description|Example|
-|------|------|-------|------|
-`api_id`|integer|ID used for the API, and unique identifier for the `locations` table|1967
-`address`|text|Address of the sign|1577 Bloor Street West
-`sign_name`|text|Name of the sign. May include address + serial number, the ward name for the Mobile WYSP, or school name for Schools WYSP|Dundas St W SB 16101191
-`dir`|text|Direction of the flow of traffic|NB
-
-### Views
-
-`report_dates` contains the list of days, for each sign, where there is more than 10 hours of data between 6:00 and 22:00.
```
Comment (ShahrzadBorjian2) [inline wys/api/README.md:89]: Cannot find report_dates in postgres

File: wys/api/README.md [hunk 9]
```diff
@@ -10,90 +16,85 @@ The script can request data on any day specified by the API call.
...
 
-`locations` contains each ID used for the API, direction of traffic the sign is gathering data for, address and name of the sign. Some signs may have information missing from any one of the fields.
-
-|Field name|Data type|Description|Example|
-|------|------|-------|------|
-`api_id`|integer|ID used for the API, and unique identifier for the `locations` table|1967
-`address`|text|Address of the sign|1577 Bloor Street West
-`sign_name`|text|Name of the sign. May include address + serial number, the ward name for the Mobile WYSP, or school name for Schools WYSP|Dundas St W SB 16101191
-`dir`|text|Direction of the flow of traffic|NB
-
```
Comment (ShahrzadBorjian2) [inline wys/api/README.md:86]: Seems a few columns are missing...

File: wys/api/README.md [hunk 10]
```diff
@@ -10,90 +16,85 @@ The script can request data on any day specified by the API call.
...
-### Views
-
-`report_dates` contains the list of days, for each sign, where there is more than 10 hours of data between 6:00 and 22:00.
+`locations` contains each ID used for the API, direction of traffic along which the sign is collecting data, and address and name of the sign. Some signs may have information missing from any one of the fields.
 
-|Field name|Data type|Description|Example|
+|Field name|Type|Description|Example|
 |------|------|-------|------|
-`api_id`|integer|ID used for the API, and unique identifier for the `locations` table|1967
```
Comment (ShahrzadBorjian2) [inline wys/api/README.md:93]: I remember seeing sign_id as well.  Do you know where and when we use it?
I think api_id is unique to each sign but sign_is is unique to sign and location.
Could you add this to this documentation?

**Review by ShahrzadBorjian2 — CHANGES_REQUESTED** [review]
Thanks a lot @tahaislam . Looks good. I just  had a few questions/ comments.

File: wys/api/README.md (hunk 4 above)
Comment (tahaislam) [reply wys/api/README.md:29]: it's exactly 5 minutes not roughly. I'll delete it

File: wys/api/README.md (hunk 3 above)
Comment (tahaislam) [reply wys/api/README.md:57]: sam as above

File: wys/api/README.md (hunk 7 above)
Comment (tahaislam) [reply wys/api/README.md:79]: the line in red was deleted in this PR. right now, it's just `speed_bins_old`

File: wys/api/README.md (hunk 9 above)
Comment (tahaislam) [reply wys/api/README.md:86]: The lines in red were deleted in this PR

File: wys/api/README.md (hunk 8 above)
Comment (tahaislam) [reply wys/api/README.md:89]: same as above

File: wys/api/README.md (hunk 10 above)
Comment (tahaislam) [reply wys/api/README.md:93]: It's the same `id` in `wys.locations`. I added a note to the README

File: wys/api/README.md [hunk 11]
```diff
@@ -26,7 +26,7 @@ The API provides data on non-regular (but uniform) 5 minute bins, i.e., at 3:44,
 
 ## Calls and Input Parameters
 
-The WYS data puller script is called by an Airflow DAG that runs daily. It collects the count of vehicles for each speed recorded by the sign in roughly 5 minute aggregate bins.
+The WYS data puller script is called by an Airflow DAG that runs daily. It collects the count of vehicles for each speed recorded by the sign in 5 minute aggregate bins.
 
 The WYS data puller script can also run independent of Airflow for specific date ranges and locations. It uses the `click` module like the `miovision` and `here` data to define input paramters. The argument to run the API is `run_api`, e.g., `python wys_api.py run_api`. It .
```
Comment (gabrielwol) [inline wys/api/README.md:31]: remove "It ." at EOL

**Comment by gabrielwol** [comment]
Just found one typo. Looks good.

**Review by gabrielwol — APPROVED** [review]
Just found one typo. Looks good.

**Pull request merged**